src/Type/Php/SprintfFunctionDynamicReturnTypeExtension.php

```diff
@@ -130,10 +138,18 @@ public function getTypeFromFunctionCall(
 
 					$singlePlaceholderEarlyReturn = $checkArgType->toString();
 				} elseif ($matches['specifier'] !== 's') {
-					$singlePlaceholderEarlyReturn = new IntersectionType([
-						new StringType(),
-						new AccessoryNumericStringType(),
-					]);
+					if ($isLowercase) {
+						$singlePlaceholderEarlyReturn = new IntersectionType([
+							new StringType(),
+							new AccessoryLowercaseStringType(),
+							new AccessoryNumericStringType(),
+						]);
+					} else {
+						$singlePlaceholderEarlyReturn = new IntersectionType([
+							new StringType(),
+							new AccessoryNumericStringType(),
+						]);
+					}
```

Comment:
maybe it would make sense to extract this "if lowercase, add lowercase-type" into a separate method, so you don't need to repeat it over and over

Reply:
Tried https://github.com/phpstan/phpstan-src/pull/3498/commits/b22c45b18b2b14316a2dbcf040d276c1c0e5bb60